contracts/rewards/src/contract/query.rs

```diff
@@ -0,0 +1,186 @@
+use cosmwasm_std::Storage;
+use error_stack::Result;
+
+use crate::{
+    error::ContractError,
+    msg::RewardsPoolResponse,
+    state::{self, Epoch, PoolId},
+};
+
+pub fn get_rewards_pool(
+    storage: &dyn Storage,
+    pool_id: PoolId,
+    block_height: u64,
+) -> Result<RewardsPoolResponse, ContractError> {
+    let pool = state::load_rewards_pool(storage, pool_id.clone())?;
+    let stored_params = state::load_params(storage);
+    let cur_epoch = Epoch::current(&stored_params, block_height)?;
+
+    let params = match state::load_epoch_tally(storage, pool_id, cur_epoch.epoch_num)? {
+        Some(epoch_tally) => epoch_tally.params,
+        None => stored_params.params,
+    };
```

Comment:
this might be a contentious take and we can discuss it with the team, but I would prefer not to reuse the stored params to make this function easier to read:
- get the pool
- get the current epoch
- try to load the tally to get the params, if that fails, load the latest params

reusing the params can potentially save you a storage access, but it gets kind of confusing to understand why which params is used

Reply:
I would rather add a comment here to do some clarification than having an extra storage access